File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        long tlen = t.length();
        long plen = p.length();
        int count = 0;
        
        for (int i = 0; i <= tlen - plen; i++) {
            String subStr = t.substring(i, i + (int)(plen));
            if (Long.parseLong(subStr) <= Long.parseLong(p)) {
                count++;
            }
        }
        
        int answer = count;
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    num = ''
    
    for i in range(0, len(t) - len(p) + 1):
        for j in range(i, i + len(p)):
            num += t[j]
        print(num)
        if (int(num) <= int(p)):
            answer += 1
        num = ''
    
    return answer